apps/src/dance/ai/DanceAiModal.tsx

```diff
@@ -790,7 +860,13 @@ const DanceAiModal: React.FunctionComponent<DanceAiModalProps> = ({
               })}
             </div>
             <div className={moduleStyles.visualizationContainer}>
-              {Array.from(Array(explanationProgress + 1).keys()).map(index => {
+              {getRangeArray(
+                BAD_GENERATED_RESULTS_COUNT - EXPLANATION_STEPS_COUNT + 1,
```

Comment:
nit: save off `BAD_GENERATED_RESULTS_COUNT - EXPLANATION_STEPS_COUNT + 1` (e.g. into some `EXPLANATION_START_INDEX`) so this can be `getRangeArray(EXPLANATION_START_INDEX, EXPLANATION_START_INDEX + explanationProgress)`?

Reply:
Good call, updated.